src/EditorFeatures/Core/InheritanceMargin/InheritanceMarginServiceHelpers.cs

```diff
@@ -3,18 +3,154 @@
+            ImmutableArray<(SymbolKey symbolKey, int lineNumber)> symbolKeyAndLineNumbers,
+            CancellationToken cancellationToken)
+        {
+            var project = solution.GetRequiredProject(projectId);
+            var compilation = await project.GetRequiredCompilationAsync(cancellationToken).ConfigureAwait(false);
+            using var _ = ArrayBuilder<SerializableInheritanceMarginItem>.GetInstance(out var builder);
+            foreach (var (symbolKey, lineNumber) in symbolKeyAndLineNumbers)
+            {
+                var symbol = symbolKey.Resolve(compilation, cancellationToken: cancellationToken).Symbol;
+                if (symbol is INamedTypeSymbol namedTypeSymbol)
+                {
+                    await GetInheritanceMemberItemsForNamedTypeAsync(solution, namedTypeSymbol, lineNumber, builder, cancellationToken);
+                }
+
+                if (symbol is IEventSymbol or IPropertySymbol or IMethodSymbol)
+                {
+                    await GetInheritanceMemberItemsForTypeMembersAsync(solution, symbol, lineNumber, builder, cancellationToken);
+                }
+            }
+
+            return builder.ToImmutable();
+        }
+
+        private static async ValueTask GetInheritanceMemberItemsForNamedTypeAsync(
+            Solution solution,
+            INamedTypeSymbol memberSymbol,
+            int lineNumber,
+            ArrayBuilder<SerializableInheritanceMarginItem> builder,
+            CancellationToken cancellationToken)
+        {
+            // Get all base types.
+            var allBaseSymbols = BaseTypeFinder.FindBaseTypesAndInterfaces(memberSymbol);
+
+            // Filter out
+            // 1. System.Object. (otherwise margin would be shown for all classes)
+            // 2. System.ValueType. (otherwise margin would be shown for all structs)
+            // 3. System.Enum. (otherwise margin would be shown for all enum)
+            var baseSymbols = allBaseSymbols
+                .WhereAsArray(symbol => symbol.SpecialType is not (SpecialType.System_Object or SpecialType.System_ValueType or SpecialType.System_Enum));
+
+            // Get all derived types
+            var derivedSymbols = await GetDerivedTypesAndImplementationsAsync(
+                solution,
+                memberSymbol,
+                cancellationToken).ConfigureAwait(false);
+
+            if (baseSymbols.Any() || derivedSymbols.Any())
+            {
+                var item = await CreateInheritanceMemberItemAsync(
+                    solution,
+                    memberSymbol,
+                    lineNumber,
+                    baseSymbols: baseSymbols.CastArray<ISymbol>(),
+                    derivedTypesSymbols: derivedSymbols.CastArray<ISymbol>(),
+                    cancellationToken).ConfigureAwait(false);
+                builder.AddIfNotNull(item);
+            }
+        }
+
+        private static async ValueTask GetInheritanceMemberItemsForTypeMembersAsync(
```

Comment:
```suggestion
        private static async ValueTask AddInheritanceMemberItemsForTypeMembersAsync(
```

Reply:
'Get' methods hsould return return something.  'Add' methods take in the collection they're going to put their items into.  can you fixup any other cases of this?

Reply:
Sure I would make sure all the names are consistent